go/libkb/chain_link.go

```diff
@@ -548,44 +587,77 @@ func (c *ChainLink) unpackStubbed(raw string, err error) error {
 	return nil
 }
 
-func (c *ChainLink) Unpack(trusted bool, selfUID keybase1.UID, packed *jsonw.Wrapper) (err error) {
-	if jw := packed.AtKey("s2"); !jw.IsNil() {
-		return c.unpackStubbed(jw.GetString())
+func (c *ChainLink) Unpack(trusted bool, selfUID keybase1.UID, packed []byte) error {
+	if s, err := jsonparser.GetString(packed, "s2"); err == nil {
+		return c.unpackStubbed(s)
 	}
 
 	tmp := ChainLinkUnpacked{}
-	tmp.sigID, err = GetSigID(packed.AtKey("sig_id"), true)
-	packed.AtKey("sig").GetStringVoid(&tmp.sig, &err)
-
-	sigVersion := 1
-	if jw := packed.AtKey("sig_version"); !jw.IsNil() {
-		sigVersion, err = jw.GetInt()
-		if err != nil {
-			return err
-		}
+	s, err := jsonparser.GetString(packed, "sig_id")
+	if err != nil {
+		return err
 	}
-	tmp.sigVersion = sigVersion
-
+	tmp.sigID, err = keybase1.SigIDFromString(s, true)
+	if err != nil {
+		return err
+	}
+	tmp.sig, err = jsonparser.GetString(packed, "sig")
 	if err != nil {
 		return err
 	}
 
-	if i, err := packed.AtKey("merkle_seqno").GetInt64(); err == nil {
+	tmp.sigVersion = 1
+	if sv, err := jsonparser.GetInt(packed, "sig_version"); err == nil {
+		tmp.sigVersion = int(sv)
+	}
+
+	if i, err := jsonparser.GetInt(packed, "merkle_seqno"); err == nil {
 		tmp.firstAppearedMerkleSeqnoUnverified = keybase1.Seqno(i)
 	}
 
-	if jw := packed.AtKey("payload_json"); !jw.IsNil() {
-		payloadJSONStr, err := jw.GetString()
+	var payload []byte
+	if trusted {
+		// use payload from sig
+		payload, err = tmp.SigPayload()
 		if err != nil {
 			return err
 		}
-		payloadJSON, err := jsonw.Unmarshal([]byte(payloadJSONStr))
+	} else {
+		// use the payload in payload_json
+		data, _, _, err := jsonparser.Get(packed, "payload_json")
```

Comment:
maybe let's not store `payload_json` in the case of V1 sigs?

Reply:
It doesn't.  See Pack().

Reply:
ok got it, thx